src/MainWindow.cpp

```diff
@@ -359,6 +359,20 @@ MainWindow::event(QEvent *event)
     return QQuickView::event(event);
 }
 
+// HACK: https://bugreports.qt.io/browse/QTBUG-83972, qtwayland cannot auto hide menu
+void
+MainWindow::mousePressEvent(QMouseEvent *event)
+{
+#if defined(Q_OS_LINUX)
+    if (QGuiApplication::platformName() == "wayland") {
+        if (event->button() == Qt::LeftButton) {
```

Comment:
I would remove the check for which button is pressed. That would make any press outside the menu close it, which should be fine too, shouldn't it?

Reply:
> I would remove the check for which button is pressed. That would make any press outside the menu close it, which should be fine too, shouldn't it?

Yes, remove the check, it still work well, Before I just worry if not check button, the menu will not show

Reply:
Understandable, I had the same concern, especially since I always confuse left and right!